File: Android/framework/src/test/java/TransactionScriptTest.java
import com.DataManipulation.BaseQueryObjectInterfaces.IBaseQueryObjectConcrete;
import com.DataSynchronizationManager;
import com.Interfaces.IDataSynchronizationManager;

import java.util.ArrayList;
import java.util.List;

import CustomerServices.Customer;
import CustomerServices.CustomerMapper;
import CustomerServices.GetCustomerByCivilStatusQuery;
import CustomerServices.GetCustomerByIdQuery;
import TransactionScripts.AlterMarriedStatusIntoSingleTS;

import static org.junit.Assert.assertEquals;

/**
 * Created by aiko on 4/8/17.
 */

public class TransactionScriptTest {
    private IDataSynchronizationManager _manager;

    @org.junit.Before
    public void Initialize()
    {
        _manager = DataSynchronizationManager.GetInstance();

        List<IBaseQueryObjectConcrete<Customer>> customerQueryObjects = new ArrayList<IBaseQueryObjectConcrete<Customer>>();

        customerQueryObjects.add(new GetCustomerByIdQuery());
        customerQueryObjects.add(new GetCustomerByCivilStatusQuery());
        _manager.RegisterEntity(Customer.class, new CustomerMapper(), customerQueryObjects);
    }

    @org.junit.Test
    public void TestRunScript()
    {
        AlterMarriedStatusIntoSingleTS transactionScript = new AlterMarriedStatusIntoSingleTS();

        transactionScript.SetInput(new GetCustomerByCivilStatusQuery.Criteria(GetCustomerByCivilStatusQuery.CivilStatus.Married));

        transactionScript.RunScript();

        List<Customer> resultsByStatus = transactionScript.GetOutput().SuccessfullyAlteredCustomers;

        assertEquals(1, resultsByStatus.size());
        assertEquals("5", resultsByStatus.get(0).Number);
        assertEquals("Test is now single", resultsByStatus.get(0).Name);
    }
}

File: DotNet/CSharp/Framework.Tests/TransactionScriptTest.cs
using System;
using System.Collections.Generic;
using Framework.Data_Manipulation;
using Framework.Tests.CustomerServices;
using Framework.Tests.TransactionScripts;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using static Framework.Tests.CustomerServices.GetCustomersByCivilStatusQuery.Criteria;

namespace Framework.Tests
{
    [TestClass]
    public class TransactionScriptTest
    {
        private IDomainObjectManager _manager = new DomainObjectManager();

        [TestInitialize]
        public void Initialize()
        {
            _manager.RegisterEntity(
                new CustomerMapper(),
                new List<IBaseQueryObject<Customer>> {
                    {new GetCustomerByIdQuery()},
                    {new GetCustomersByCivilStatusQuery()}
                });
        }

        [TestMethod]
        public void TestRunScript()
        {
            AlterMarriedStatusIntoSingleTS transactionScript = new AlterMarriedStatusIntoSingleTS(_manager);

            transactionScript.Input = GetCustomersByCivilStatusQuery.Criteria.SearchByStatus(CivilStatus.Married);

            transactionScript.RunScript();

            IList<Customer> resultsByStatus = transactionScript.Output.SuccessfullyAlteredCustomers;

            Assert.AreEqual(2, resultsByStatus.Count);
            Assert.AreEqual("5", resultsByStatus[0].Number);
            Assert.AreEqual("Test is now single", resultsByStatus[0].Name);
            Assert.AreEqual("7", resultsByStatus[1].Number);
            Assert.AreEqual("Test is now single", resultsByStatus[1].Name);

        }
    }
}
